tools/SourceKit/tools/sourcekitd/lib/API/Requests.cpp

```diff
@@ -2072,6 +2081,11 @@ class SKEditorConsumer : public EditorConsumer {
   bool handleSourceText(StringRef Text) override;
   bool handleSerializedSyntaxTree(StringRef Text) override;
   bool syntaxTreeEnabled() override { return Opts.EnableSyntaxTree; }
+
+  bool syntaxReuseInfoEnabled() override { return Opts.EnableSyntaxReuseInfo; }
+  bool handleSyntaxReuseRegions(
+      std::vector<std::pair<unsigned, unsigned>> ReuseRegions) override;
```

Comment:
Define a struct instead of using `std::pair<unsigned, unsigned>`. In general, avoid using `std::pair` in a non-local context since we don't get any interesting information from it. For instance, are these `unsigned` offset, line-column, or something else?

Reply:
I wanted to do that, but the data is generated in swiftParse and used in SourceKit. Neither one imports the other (which I believe is a good thing), so the question comes up where this new type should be defined or if we want to convert between two different types in soucekitd. I decided that using `std::pair` is the lesser of three evils but am open to better suggestions.